TCAT/Cells/ServiceAlertTableViewCell.swift

```diff
@@ -0,0 +1,237 @@
+                make.height.equalTo(timeSpanLabel.intrinsicContentSize.height)
+            }
+        } else {
+            timeSpanLabel.snp.remakeConstraints { (make) in
+                make.top.leading.trailing.equalToSuperview().inset(borderInset)
+                make.height.equalTo(timeSpanLabel.intrinsicContentSize.height)
+            }
+        }
+        
+        descriptionLabel.snp.remakeConstraints { (make) in
+            make.top.equalTo(timeSpanLabel.snp.bottom).offset(12)
+            make.leading.equalTo(timeSpanLabel)
+            make.trailing.equalToSuperview().inset(borderInset)
+            if let text = descriptionLabel.text {
+                let width = contentView.frame.width - (CGFloat)(2 * borderInset)
+                let heightValue = ceil(text.heightWithConstrainedWidth(width: width, font: descriptionLabel.font))
+                make.height.equalTo(ceil(heightValue))
+            } else {
+                make.height.equalTo(descriptionLabel.intrinsicContentSize.height)
+            }
+        }
+        
+        if let stackView = affectedRoutesStackView {
+            // When both a separator view and stackView are required
+            affectedRoutesLabel.snp.remakeConstraints { (make) in
+                make.leading.equalTo(timeSpanLabel)
+                make.top.equalTo(descriptionLabel.snp.bottom).offset(24)
+                make.width.equalTo(affectedRoutesLabel.intrinsicContentSize.width)
+                make.height.equalTo(affectedRoutesLabel.intrinsicContentSize.height)
+            }
+            stackView.snp.remakeConstraints { (make) in
+                make.top.equalTo(affectedRoutesLabel.snp.bottom).offset(8)
+                make.leading.equalTo(timeSpanLabel)
+                make.trailing.bottom.equalToSuperview().inset(borderInset)
+            }
+        } else {
+            descriptionLabel.snp.makeConstraints { (make) in
+                make.bottom.equalToSuperview().inset(borderInset)
+            }
+        }
+        super.updateConstraints()
+    }
+    
+    private func formatTimeString(_ fromDate: String, toDate: String) -> String {
+        
+        let newformatter = DateFormatter()
+        newformatter.dateFormat = "yyyy-MM-dd'T'HH:mm:ss.sZZZZ"
+        newformatter.locale = Locale(identifier: "en_US_POSIX")
+        
+        let fromDate = newformatter.date(from: fromDate)
+        let toDate = newformatter.date(from: toDate)
+
+        let formatter = DateFormatter()
+        formatter.dateFormat = "EEEE M/d"
+
+        if let unWrappedFromDate = fromDate, let unWrappedToDate = toDate {
+            let formattedFromDate = formatter.string(from: unWrappedFromDate)
+            let formattedToDate = formatter.string(from: unWrappedToDate)
+            
+            return "\(formattedFromDate) - \(formattedToDate)"
+        }
+        
+        return "Time: Unknown"
+    }
+    
+    private func calculateMaxIconsPerRow() -> Int {
```

Comment:
Maybe this should just be a computed property, but then you have to recalculate it each time. @AAAstorga Thoughts on making a layout-based calculation like this be a function or computed property? If you need more context, poke around or ask